src/tests/builtins.rs

```diff
@@ -256,6 +244,240 @@ mod attr_values {
     }
 }
 
+mod baseNameOf {
+    use super::*;
+
+    // Returns everything following the final slash
+
+    #[test]
+    fn eval() {
+        assert_eq!(
+            eval_ok("builtins.baseNameOf \"/foo/bar/baz\""),
+            Value::Str("baz".into())
+        );
+        assert_eq!(
+            eval_ok("builtins.baseNameOf \"/foo/bar/baz/\""),
+            Value::Str("".into())
```

Comment:
Nix seems to return `"baz"` for this test:
```bash
% nix eval --expr 'builtins.baseNameOf "/foo/bar/baz/"'
"baz"
```

```suggestion
            Value::Str("baz".into())
```